File: Languages/R/tasklist.r
tarefas <- list()

menu <- function() {
  cat("\nListador de Tarefas\n")
  cat("1. Adicionar tarefa\n")
  cat("2. Listar tarefas\n")
  cat("3. Remover tarefa\n")
  cat("4. Sair\n")
  cat("Escolha uma opção: ")
}

adicionar_tarefa <- function() {
  tarefa <- readline(prompt = "Digite a descrição da tarefa: ")
  tarefas <<- c(tarefas, tarefa)
  cat("Tarefa adicionada.\n")
}

listar_tarefas <- function() {
  if (length(tarefas) == 0) {
    cat("Nenhuma tarefa cadastrada.\n")
  } else {
    cat("Tarefas:\n")
    for (i in seq_along(tarefas)) {
      cat(i, ": ", tarefas[[i]], "\n", sep = "")
    }
  }
}

remover_tarefa <- function() {
  listar_tarefas()
  if (length(tarefas) > 0) {
    indice <- as.integer(readline(prompt = "Digite o número da tarefa a ser removida: "))
    if (indice > 0 && indice <= length(tarefas)) {
      tarefas <<- tarefas[-indice]
      cat("Tarefa removida.\n")
    } else {
      cat("Índice inválido.\n")
    }
  }
}

executar_listador <- function() {
  repeat {
    menu()
    opcao <- as.integer(readline())
    if (opcao == 1) {
      adicionar_tarefa()
    } else if (opcao == 2) {
      listar_tarefas()
    } else if (opcao == 3) {
      remover_tarefa()
    } else if (opcao == 4) {
      cat("Saindo...\n")
      break
    } else {
      cat("Opção inválida. Tente novamente.\n")
    }
  }
}

executar_listador()

File: Languages/Perl/tasklist.pl
use strict;
use warnings;

my @tarefas;

while (1) {
    print "\nListador de Tarefas\n";
    print "1. Adicionar tarefa\n";
    print "2. Listar tarefas\n";
    print "3. Remover tarefa\n";
    print "4. Sair\n";
    print "Escolha uma opção: ";

    my $opcao = <STDIN>;
    chomp($opcao);

    if ($opcao == 1) {
        print "Digite a descrição da tarefa: ";
        my $descricao = <STDIN>;
        chomp($descricao);
        push @tarefas, $descricao;
        print "Tarefa adicionada.\n";
    } elsif ($opcao == 2) {
        if (@tarefas) {
            print "Tarefas:\n";
            for my $i (0 .. $#tarefas) {
                print "$i: $tarefas[$i]\n";
            }
        } else {
            print "Nenhuma tarefa cadastrada.\n";
        }
    } elsif ($opcao == 3) {
        if (@tarefas) {
            print "Digite o número da tarefa a ser removida: ";
            my $indice = <STDIN>;
            chomp($indice);
            if ($indice >= 0 && $indice < @tarefas) {
                splice(@tarefas, $indice, 1);
                print "Tarefa removida.\n";
            } else {
                print "Índice inválido.\n";
            }
        } else {
            print "Nenhuma tarefa para remover.\n";
        }
    } elsif ($opcao == 4) {
        print "Saindo...\n";
        last;
    } else {
        print "Opção inválida. Tente novamente.\n";
    }
}